src/transport/emitter.ts

```diff
@@ -97,7 +88,7 @@ export class Emitter extends EventEmitter {
    * @param {Function} listener to call on event
    * @return {void}
    */
-  static on(event: "cloudevent" | "newListener" | "removeListener", listener: (...args: any[]) => void): void {
```

Comment:
I liked the idea of typing more precisely than the generic string, I checked to revert this on your branch, and seems ok